### PR TITLE
Refactored deliverables to component cards

### PR DESCRIPTION
**Pull request by tommylau523**
Refactored out deliverable views so that application portal can use cards and inject different buttons and navbar accordingly.

This is in preparation for implementing application deliverable screen down the line.

Some design notes:
- Crumbs, navigation are injected on the router, in this case `DeliverablePage`. This is so when Application portal implements deliverable views, it can look different.

### CODE REVIEW

**Comment by tommylau523** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2897?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2897** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2897?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2895** <a href="https://app.graphite.dev/github/pr/terraware/terraware-web/2895?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @tommylau523 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: photo-data/2024/07/24/species-list-deliverable/235717-9187E573430C5A91.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Project ID,Species ID,Status,Rationale,Feedback,Internal Comment,Native / Non-Native,Species Scientific Name,Species Common Name
+1,36,Approved,abcdefg,,,,Abutilon eremitopetalum,COMMON NAME
+1,37,Approved,True,,,,Abutilon incanum,"hoary abutilon, Ma'o [Mao]"
```
Comment (joshbuchea) [inline photo-data/2024/07/24/species-list-deliverable/235717-9187E573430C5A91.csv:3]: Not sure if this was added unintentionally?

**Review by joshbuchea — APPROVED** [review]
Whew this was a big refactor! 😅 Changes look good 👍

**Comment by tommylau523** [comment]
### Merge activity

* **Jul 25, 6:52 PM EDT**: @tommylau523 merged this pull request with [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-web/2897).

**Pull request merged**